central/declarativeconfig/updater/auth_provider_updater.go

```diff
@@ -75,17 +76,30 @@ func (u *authProviderUpdater) DeleteResources(ctx context.Context, resourceIDsTo
 	var authProviderDeletionErr *multierror.Error
 	var authProviderIDs []string
 	for _, authProvider := range authProviders {
-		if err := u.authProviderRegistry.DeleteProvider(ctx, authProvider.GetId(), true, true); err != nil {
+		referencingGroups, err := u.groupDS.GetFiltered(ctx, func(group *storage.Group) bool {
+			return group.GetProps().GetAuthProviderId() == authProvider.GetId()
+		})
+		if err != nil || len(referencingGroups) > 0 {
+			if err == nil {
+				err = errox.ReferencedByAnotherObject.Newf("auth provider is still referenced by groups")
+			}
```

Comment:
Less indentations and also let's provide the number of groups referenced and the auth provider name in the error message: 
```suggestion
		if len(referencingGroups) > 0 {
			err = errox.ReferencedByAnotherObject.Newf("auth provider %s is still referenced by %d groups", authProvider.GetName(), len(referencingGroups))
		}
		if err != nil {
```